resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -705,6 +705,19 @@ private[spark] class Client(
       }
     }
 
+    val confDir =
+      sys.env.getOrElse("SPARK_CONF_DIR", sys.env("SPARK_HOME") + File.separator + "conf")
+    val dir = new File(confDir)
+    if (dir.isDirectory) {
+      val files = dir.listFiles(new FileFilter {
+        override def accept(pathname: File): Boolean = {
+          pathname.isFile && pathname.getName.endsWith("xml")
```

Comment:
Shall we explicitly match the file name, like "hive-site.xml"? Looks like only check file name ends with "xml" will also include other unwanted files indefinitely.

Reply:
According to the doc, 
> Configuration of Hive is done by placing your hive-site.xml, core-site.xml (for security configuration), and hdfs-site.xml (for HDFS configuration) file in conf/.
here we may not only get hive-site.xml

Reply:
Yes, I understand. My question is that do we need to explicitly check the expected file names, rather than blindly match any xml file?

Reply:
I guess that we do not check the `$HADOOP(YARN)_CONF_DIR` either